synapse/handlers/sync.py

```diff
@@ -1093,6 +1093,9 @@ async def generate_sync_result(
         one_time_key_counts: JsonDict = {}
         unused_fallback_key_types: List[str] = []
         if device_id:
+            # TODO: If this is an incremental sync and there's been no change in a count
+            # since the provided since token, then simply omit that algorithm's entry
+            # altogether.
```

Comment:
I'm not 100% convinced this is a thing we want to do, per https://github.com/matrix-org/synapse/issues/10484#issuecomment-887415854.

Reply:
Thanks for calling this out. Shall I remove this TODO for now as we design a more robust signalling solution on the spec side?

Reply:
I dunno, maybe soften the phrasing a bit: "consider omitting the entry altogether". possibly also link to the discussion in #10484.

Reply:
(or the spec issue)

Reply:
Out of that discussion came an issue on matrix-doc. I've changed the comment here to link to it in the meantime.